### PR TITLE
Added os_routers_facts module

### PR DESCRIPTION
**Pull request by greg-hellings**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

os_routers_facts
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = /home/ghelling/build/redhat/ci/stachyon/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Adds a new module, modeled on other os_*_facts.py, that fetches information about configured routers in an OpenStack instance.

Pulls and registers information for regarding the list of
routers in an OpenStack installation

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @greg-hellings. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_routers_facts.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_routers_facts
+short_description: Retrieve facts about one or more OpenStack routers.
+version_added: "2.2"
+author: "Davide Agnello (@dagnello)"
```
Comment (Shrews) [inline cloud/openstack/os_routers_facts.py]: This should be your name and Github idea.

File: cloud/openstack/os_routers_facts.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            description: Network shared flag.
+            returned: success
+            type: boolean
+'''
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=False, default=None),
+        filters=dict(required=False, default=None)
```
Comment (Shrews) [inline cloud/openstack/os_routers_facts.py]: Need to specify type for non-string parameters.

**Comment by rcarrillocruz** [comment]
Also, please rename this to os_router_facts.
We use singular in other modules, so better follow that pattern.

**Comment by rcarrillocruz** [comment]
needs_revision

File: cloud/openstack/os_routers_facts.py (hunk 1 above)
Comment (Shrews) [inline cloud/openstack/os_routers_facts.py]: s/idea/ID/

Comment (greg-hellings) [inline cloud/openstack/os_routers_facts.py]: Whoops! Fixed.

File: cloud/openstack/os_routers_facts.py (hunk 2 above)
Comment (greg-hellings) [inline cloud/openstack/os_routers_facts.py]: Fixed.

**Comment by greg-hellings** [comment]
@rcarrillocruz I copied the naming convention from os_networks_facts and os_subnets_facts. I've updated this module to match the singular naming convention.

**Comment by rcarrillocruz** [comment]
@greg-hellings yeah, i know there is no consistency :/ , we are trying to converge on singular on future modules.

Thanks

File: cloud/openstack/os_router_facts.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    filters:
+      tenant_id: 55e2ce24b2a245b09f181bf025724cbe
+      subnets:
+        - 057d4bdf-6d4d-4728-bb0f-5ac45a6f7400
+        - 443d4dc0-91d4-4998-b21c-357d10433483
```
Comment (Shrews) [inline cloud/openstack/os_router_facts.py:80]: This is not a valid example. Looks like you just copied/pasted from os_networks_facts.py.

File: cloud/openstack/os_router_facts.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            type: list of strings
+        tenant_id:
+            description: Tenant id associated with this network.
+            returned: success
+            type: string
+        shared:
+            description: Network shared flag.
+            returned: success
+            type: boolean
+'''
```
Comment (Shrews) [inline cloud/openstack/os_router_facts.py:114]: These return values are not valid as you simply copied/pasted from the network facts module.

**Comment by Shrews** [comment]
Mostly ok. Still needs some work where noted.

**Pull request closed without merging**